content/blog/scale-kubernetes-cluster-using-hpe-greenlake-terraform-provider.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Scale Kubernetes Cluster using HPE GreenLake Terraform Provider
+date: 2022-07-12T14:06:02.781Z
+author: Arshad Saquib C L
+authorimage: /img/mypic192x192.jpg
+tags:
+  - " hpe-greenlake"
+  - devops
+  - terraform
+  - open source
+---
+The process of managing and provisioning computer data centers through machine-readable definition files, also known as Infrastructure-as-Code (IaC), offers many significant benefits. It helps to increase operational agility, simplify management, reduce errors, and save cost.
+
+In this post, we will explore options to declare and scale Kubernetes clusters on HPE GreenLake using the HPE GreenLake Terraform Provider.
+
+# Prerequisite
+
+See the blog post [Kubernetes Cluster as Code - Part 1](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/) that includes steps for creating a Kubernetes cluster. This post expands upon the scenario by examining how to scale a cluster.
+
+# Verify existing Kubernetes cluster
+
+After the cluster is created following the [Kubernetes Cluster as Code - Part 1 blog post](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), launch HPE GreenLake Central console and verify that the cluster is present under the appropriate tenant.
+
+You should see the **tf-test** cluster present under **Dashboard -> Manage your Private Cloud -> Containers**.
+
+![](/img/cluster_list.jpg)
+
+Below is the reference Terraform configuration file for the existing cluster.
+
+```hcl
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+ 
+variable "HPEGL_SPACE" {
+  type = string
+}
+ 
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = var.HPEGL_SPACE
+ }
+ 
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+ 
+resource hpegl_caas_cluster test {
+  name         = "tf-test"
+  blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+  site_id      = data.hpegl_caas_site.blr.id
+  space_id     = var.HPEGL_SPACE
+}
+```
+
+# Add worker node to Kubernetes cluster resource
+
+You can scale the cluster by adding a worker node. The following **worker_nodes** attributes are specified to add or modify node pools in the declared Kubernetes cluster resource.
```

Comment:
Grammatically, it should be "The following worker-node attributes are specified..."